source/agora/consensus/protocol/Nominator.d

```diff
@@ -1062,6 +1055,45 @@ extern(D):
         log.trace("valueExternalized: added slot id {} to ledger at height {}", height, last_height);
         () @trusted { this.fully_validated_value.clear(); }();
         () @trusted { this.seen_envs.clear(); }();
+        this.armTaskTimer(TimersIdx.Externalize, ExternalizeTaskDelay);
+    }
+
+    /// If we have majority signatures then externalize to the ledger otherwise try again later
+    extern(D) private void checkExternalize () nothrow
```

Comment:
I dont think this task is really necessary. I feel like we can use `updateMultiSignature` as a checkpoint to determine if we can externalize the pending block.

Reply:
Good idea, I'll give it a go.

Reply:
Updated.